controller/controller_server/csi_controller_server.py

```diff
@@ -387,9 +387,34 @@ def ControllerUnpublishVolume(self, request, context):
 
     def ValidateVolumeCapabilities(self, request, context):
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
+            with get_agent(user, password, array_addresses, array_type).get_mediator() as array_mediator:
+                volume = array_mediator.get_object_by_id(object_id=volume_id, object_type=config.VOLUME_TYPE_NAME)
+
+            if request.volume_context:
+                utils.validate_volume_context_match_volume(request.volume_context, volume)
```

Comment:
so if `request.volume_context` is `None`, why should `get_object_by_id` be called at all?

Reply:
for not found